radar_parlamentar/importadores/conv.py

```diff
@@ -118,180 +125,44 @@ def _gera_votos(self, votacao, nome_partido, opcoes):
-
-    def _gera_votacao7(self):
-
-        numero_proposicao = '7'
-        descricao_proposicao = 'Aumento do efetivo militar'
-        prop = self._gera_proposicao(numero_proposicao, descricao_proposicao)
-        votacao = self._gera_votacao(numero_proposicao, descricao_proposicao,
-                                     DATA_NO_SEGUNDO_SEMESTRE, prop)
-
-        votos_girondinos = [models.SIM, models.SIM, models.ABSTENCAO]
-        self._gera_votos(votacao, GIRONDINOS, votos_girondinos)
-
-        votos_jacobinos = [models.SIM, models.SIM, models.SIM]
-        self._gera_votos(votacao, JACOBINOS, votos_jacobinos)
-
-        votos_monarquistas = [models.SIM, models.AUSENTE, models.SIM]
-        self._gera_votos(votacao, MONARQUISTAS, votos_monarquistas)
-
-    # votação com atributos diferentes para teste
-    def _gera_votacao8(self):
-
-        numero_proposicao = '8'
-        descricao_proposicao = 'Guerra contra a Inglaterra'
-        prop = models.Proposicao()
-        prop.id_prop = numero_proposicao
-        prop.sigla = 'PL'
-        prop.numero = numero_proposicao
-        prop.ementa = 'o uso proibido de armas químicas'
-        prop.descricao = 'descricao da guerra'
-        prop.casa_legislativa = self.casa
-        prop.indexacao = 'bombas, efeitos, destruições'
-        prop.save()
-        votacao = self._gera_votacao(numero_proposicao, descricao_proposicao,
-                                     DATA_NO_SEGUNDO_SEMESTRE, prop)
-
-        votos_girondinos = [models.NAO, models.NAO, models.NAO]
-        self._gera_votos(votacao, GIRONDINOS, votos_girondinos)
-
-        votos_jacobinos = [models.ABSTENCAO, models.NAO, models.NAO]
-        self._gera_votos(votacao, JACOBINOS, votos_jacobinos)
-
-        votos_monarquistas = [models.SIM, models.AUSENTE, models.SIM]
-        self._gera_votos(votacao, MONARQUISTAS, votos_monarquistas)
-
-    def _gera_votacao9(self):
-        numero_proposicao = '9'
-        descricao_proposicao = 'Contratar médicos para a capital'
-        prop = self._gera_proposicao(numero_proposicao, descricao_proposicao)
-        votacao = self._gera_votacao(numero_proposicao, descricao_proposicao,
-                                     DATA_VOTACAO_9, prop)
-
-        votos_girondinos = [models.SIM, models.SIM, models.ABSTENCAO]
-        self._gera_votos(votacao, GIRONDINOS, votos_girondinos)
-
-        votos_jacobinos = [models.SIM, models.SIM, models.SIM]
-        self._gera_votos(votacao, JACOBINOS, votos_jacobinos)
-
-        votos_monarquistas = [models.SIM, models.AUSENTE, models.SIM]
-        self._gera_votos(votacao, MONARQUISTAS, votos_monarquistas)
+    def _obtem_dados_json(self):
+        data = None
+        with open(os.path.join(ARQUIVO, 'votacoes.json')) as arquivo:
+            data = json.load(arquivo)
+        return data
+
+    def _gera_votacao_geral(self):
+        data = self._obtem_dados_json()
+        votacoes_params = data["votacoes_params"]
```

Comment:
A variável `data`,  gerada na linha anterior pela chamada do método `_obtem_dados_json()` **pode** ser `None`.
Se for, a expressão `data["votacoes_params"]` vai levantar uma excessão necessariamente. Isso precisa ser tratado corretamente, e isso pode se dar de diversas formas.

Verificando se `data is None`, retornando um dicionário com valor vazio para a chave "votacoes_params", retornando um dicionário vazio no método e usando o `get` ao invésde `["votacoes_params"]` e definindo um valor default no get... enfim...

Independente disso, eu acho que estão faltando testes para os novos métodos que foram escritos/implementados.

Reply:
Ok! Irei fazer as devidas alterações. Obrigado pelas dicas =)